examples/panel/book-recommender/rag_book_recommender.py

```diff
@@ -0,0 +1,89 @@
+import os
+import json
+import pickle
+import pandas as pd
+
+from pathlib import Path
+from openai import OpenAI
+
+client = OpenAI()
+
+file_path = os.path.join('assets', 'goodreads.csv')
+df = pd.read_csv(file_path)
+
+
+file_path = os.path.join('assets', 'title_to_description.pkl')
+with open(file_path, 'rb') as file:
+    DESCRIPTIONS = pickle.load(file)
+
+
+def get_authors():
+    file_path = os.path.join('assets', 'author_to_title.pkl')
+    with open(file_path, 'rb') as file:
+        authors = pickle.load(file)
+    return authors
+
+
+def get_embeddings():
```

Comment:
the overall code is a bit hard to follow because there are no docstrings. what's in the embeddings.json? I'm guessing the keys are the titles and the values are the embeddings produced from the description?

Reply:
[fixed](https://github.com/ploomber/doc/pull/127/files#diff-99d50d468b15d576674758b1fbde220b390f380dc2398dd8b4024a04218dd2ccR36-R38)